### DISCUSSION TITLE
Custom build command

### DISCUSSION THREAD

**Discussion by xxSkyy** [ElMassimo/vite_ruby#112] [Q&A] [answered]
Hi, on one of my low memory vps building with vite throws `Allocation failed - JavaScript heap out of memory` error, to fix it I need to add `NODE_OPTIONS=--max-old-space-size=4000` and run it before `rails assets:precompile`.

Are there option to customize build command on assets:precompile with properties like `NODE_OPTIONS=` >

**Comment by ElMassimo** [comment, marked as answer]
Hi Sky! @xxSkyy 

Setting the `NODE_OPTIONS` environment variable to be globally available in the server that performs the assets precompilation should make it available in the Vite process as well.

An alternative would be to customize [`viteBinPath`](https://vite-ruby.netlify.app/config/#vitebinpath) in your `config/vite.json` to use a different executable (for example, one that executes `vite` after setting `NODE_OPTIONS`).

**Reply by xxSkyy** [reply]
Thanks for reply, I'll figure it out.
Also are there option to run vite dev with `TAILWIND_MODE=watch` ? 
I need it to proper JIT mode  `TAILWIND_MODE=watch bin/vite dev` doesn't take arguments

**Reply by pocheptsov** [reply]
@xxSkyy we set it in the `tailwind.config.js` configuration file:
```
module.exports = {
  mode: 'jit',
  ...
```

**Reply by xxSkyy** [reply]
> @xxSkyy we set it in the `tailwind.config.js` configuration file:
> 
> ```
> module.exports = {
>   mode: 'jit',
>   ...
> ```

But I've got problem with JIT exact this one
https://tailwindcss.com/docs/just-in-time-mode#styles-don-t-update-when-saving-content-files
and I need set TAILWIND_MODE , then JIT mode works well

**Reply by ElMassimo** [reply]
Try asking in the Vite.js Discord forum, there must be other users using Tailwind.

As far as the vite binary is concerned, any environment variables you provide will be passed through, so it's likely that the docs you linked to are referring to the PostCSS cli, and don't apply in this case.

I prefer [Windi CSS](https://vite-ruby.netlify.app/guide/plugins.html#windi-css), which is mostly compatible with Tailwind but integrates very nicely with Vite.js

Good luck!

**Reply by xxSkyy** [reply]
> Try asking in the Vite.js Discord forum, there must be other users using Tailwind.
> 
> As far as the vite binary is concerned, any environment variables you provide will be passed through, so it's likely that the docs you linked to are referring to the PostCSS cli, and don't apply in this case.
> 
> I prefer [Windi CSS](https://vite-ruby.netlify.app/guide/plugins.html#windi-css), which is mostly compatible with Tailwind but integrates very nicely with Vite.js
> 
> Good luck!

But it's problem with vite_ruby, I can't pass anything to vite server with command, example

Vite ruby:
```
yarn vite --port 2222 --host

  vite v2.4.4 dev server running at:

  > Local: http://localhost:3036/vite-dev/
  > Network: use `--host` to expose

  ready in 145ms.
```

Normal Vite:
```
yarn vite --port 2222 --host

  vite v2.4.1 dev server running at:

  > Local:    http://localhost:2222/
  > Network:  http://10.0.0.51:2222/

  ready in 176ms.
```

**Reply by ElMassimo** [reply]
@xxSkyy The `bin/vite` executable is not meant to be used through `yarn`—use it directly as `bin/vite dev`.

`host` and `port` should not be passed to the CLI—configure them using [`config/vite.json`](https://vite-ruby.netlify.app/config/#shared-configuration-file-%F0%9F%93%84). Else, the Ruby process will be configured differently and won't be able to connect to the Vite dev server.

<hr/>

Regardless of the above, using `yarn` `1.22.10` I am able to pass environment variables and CLI options, such as:

```
DEBUG=vite* yarn vite dev --port 2222

  vite v2.4.2 dev server running at:

  > Local: http://localhost:2222/
  > Network: use `--host` to expose

  ready in 276ms.
```

**Reply by ElMassimo** [reply]
@xxSkyy The problem you are experiencing with Tailwind seems related to the internal usage of the [`base`](https://vitejs.dev/config/#base) configuration. See https://github.com/ElMassimo/vite_ruby/discussions/115 for more details.

As a temporary workaround, you could [downgrade `vite` to `2.3.8`](https://github.com/tailwindlabs/tailwindcss/issues/4978#issuecomment-897620776), it seems like the problem with `base` was introduced in one of the recent Vite versions.
